File: android/src/main/kotlin/com/exponea/data/ConsentEncoder.kt
package com.exponea.data

import com.exponea.sdk.models.Consent

class ConsentEncoder {
    companion object {
        fun encode(consent: Consent): Map<String, Any?> {
            return mapOf(
                "id" to consent.id,
                "legitimateInterest" to consent.legitimateInterest,
                "sources" to mapOf(
                    "createdFromCRM" to consent.sources.createdFromCRM,
                    "imported" to consent.sources.imported,
                    "fromConsentPage" to consent.sources.fromConsentPage,
                    "privateAPI" to consent.sources.privateAPI,
                    "publicAPI" to consent.sources.publicAPI,
                    "trackedFromScenario" to consent.sources.trackedFromScenario
                ),
                "translations" to consent.translations
            )
        }
    }
}

File: ios/Classes/Data/ConsentEncoder.swift
//
//  ConsentEncoder.swift
//  exponea
//

import Foundation
import ExponeaSDK

class ConsentEncoder {
    static func encode(_ consent: Consent) -> [String:Any?] {
        return [
            "id": consent.id,
            "legitimateInterest": consent.legitimateInterest,
            "sources": [
                "createdFromCRM": consent.sources.isCreatedFromCRM,
                "imported": consent.sources.isImported,
                "fromConsentPage": consent.sources.isFromConsentPage,
                "privateAPI": consent.sources.privateAPI,
                "publicAPI": consent.sources.publicAPI,
                "trackedFromScenario": consent.sources.isTrackedFromScenario
            ],
            "translations": consent.translations
        ]
    }
}
